bears/ruby/RubySmellBear.py

```diff
@@ -0,0 +1,162 @@
+        return '--format', 'json', '-c', config_file
+
+    def process_output(self, output, filename, file):
+        output = json.loads(output) if output else []
+        for issue in output:
+            sourceranges = []
+            for line in issue['lines']:
+                sourceranges.append(SourceRange.from_values(
+                    file=filename, start_line=line))
+
+            yield Result(
+                origin="{} ({})".format(self.__class__.__name__,
+                                        issue['smell_type']),
+                message="{} (in context {}) {}.".format(issue['name'],
+                                                        issue['context'],
+                                                        issue['message']),
+                affected_code=sourceranges,
+                additional_info="More information is available at {}"
+                                ".".format(issue["wiki_link"]))
+
+    def generate_config(self,
+                        attribute: bool=True,
+                        boolean_parameter: bool=True,
+                        class_variable: bool=True,
+                        control_parameter: bool=True,
+                        data_clump: bool=True,
+                        duplicate_method_call: bool=True,
+                        feature_envy: bool=True,
+                        missing_module_description: bool=True,
+                        long_param_list: bool=True,
+                        long_yield_list: bool=True,
+                        module_initialize: bool=True,
+                        nested_iterators: bool=True,
+                        nil_check: bool=True,
+                        prima_donna_method: bool=True,
+                        repeated_coditional: bool=True,
+                        too_many_instance_variables: bool=True,
+                        too_many_methods: bool=True,
+                        too_many_statements: bool=True,
+                        uncommunicative_method_name: bool=True,
+                        uncommunicative_module_name: bool=True,
+                        uncommunicative_param_name: bool=True,
+                        uncommunicative_var_name: bool=True,
+                        unused_params: bool=True,
+                        unused_private_method: bool=False,
+                        utility_function: bool=True):
+        """
+        :param attribute:
+            Warns if a setter in a class is detected.
+        :param boolean_parameter:
+            Warns if a boolean parameter in a function is detected (control
+            coupling).
+        :param class_variable:
+            Warns if a class variable is detected.
+        :param control_parameter:
+            Warns if a parameter controls the function behaviour (control
+            coupling).
+        :param data_clump:
+            Warns when the same two or three items frequently appear together.
+        :param duplicate_method_call:
```

Comment:
We **should** have a Generic Bear for this. It's actually a very simple thing to check right ? Basically see if a set of lines appear more than once in a code base.
There should be a good algorithm which can do this (It comes under substring search problems i think). If my understanding is right, could you make an issue please ?